src/cli/project/export/pypi_requirements.rs

```diff
@@ -0,0 +1,323 @@
+        //
+        // remove "direct+ since not valid for pip urls"
+        let s = s.trim_start_matches("direct+");
+
+        let hash_flag = if include_hash {
+            get_pypi_hash_str(p.data().package)
+        } else {
+            None
+        };
+
+        Self {
+            source: s.to_string(),
+            hash_flag,
+            editable: p.is_editable(),
+        }
+    }
+
+    fn to_req_entry(&self) -> String {
+        let mut entry = String::new();
+
+        if self.editable {
+            entry.push_str("-e ");
+        }
+        entry.push_str(&self.source);
+
+        if let Some(hash) = &self.hash_flag {
+            entry.push_str(&format!(" {}", hash));
+        }
+
+        entry
+    }
+}
+
+fn get_pypi_hash_str(package_data: &PypiPackageData) -> Option<String> {
+    if let Some(hashes) = &package_data.hash {
+        let h = match hashes {
+            PackageHashes::Sha256(h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5Sha256(_, h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5(h) => format!("--hash=md5:{:x}", h).to_string(),
+        };
+        Some(h)
+    } else {
+        None
+    }
+}
+
+fn render_pypi_requirements(
+    target: impl AsRef<Path>,
+    packages: &[PypiPackageReqData],
+) -> miette::Result<()> {
+    if packages.is_empty() {
+        return Ok(());
+    }
+
+    let target = target.as_ref();
+
+    let reqs = packages
+        .iter()
+        .map(|p| p.to_req_entry())
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    fs::write(target, reqs)
+        .into_diagnostic()
+        .with_context(|| format!("failed to write requirements file: {}", target.display()))?;
+
+    Ok(())
+}
+
+fn render_env_platform(
+    output_dir: &Path,
+    env_name: &str,
+    env: &Environment,
+    platform: &Platform,
+    split_nohash: bool,
+) -> miette::Result<()> {
+    let packages = env.packages(*platform).ok_or(miette::miette!(
+        "platform '{platform}' not found for env {}",
+        env_name,
+    ))?;
+
+    let mut pypi_packages: Vec<PypiPackageReqData> = Vec::new();
+
+    for package in packages {
+        match package {
+            Package::Pypi(p) => pypi_packages.push(PypiPackageReqData::from_pypi_package(&p)),
+            Package::Conda(cp) => {
+                tracing::warn!(
+                    "ignoring Conda package {} since Conda packages are not supported in requirements.txt",
+                    cp.package_record().name.as_normalized()
+                );
+            }
+        }
+    }
+
+    let (base, nohash) = if split_nohash {
+        // Split package list based on presence of hash since pip currently treats requirements files
+        // containing any hashes as if `--require-hashes` has been supplied. The only known workaround
+        // is to split the dependencies, which are typically from vcs sources into a separate
+        // requirements.txt and to install it separately.
```

Comment:
I had to read this part to understand the flag. Could you incorporate this information in the documentation and help of the flag?